src/Components/Admin/AdminDashboard.tsx

```diff
@@ -6,14 +6,50 @@ import WorkshopEdit from "../Workshop/WorkshopEdit";
 import CardTile from "../Workshop/CardTile";
 import Loading from "../Layout/Loading";
 import { Row, Col, Container, Alert } from "react-bootstrap";
+import { workshop, studentsAtWorkshop } from "../Firebase/interface";
 import FileSaver from "file-saver";
 
+interface AdminDashboardProps {
+  workshop_data: workshop[];
+  student_data: studentsAtWorkshop[];
+  readStudentData(): void;
+  readWorkshopData(): void;
+  updateWorkshop(workshopID: string, workshopObject: workshop): void;
+  updateLevel(workshopID: string, level: number): void;
+  createWorkshop(workshopObject: workshop): void;
+  deleteWorkshop(workshopID: string): void;
+  updateStatus(workshopID: string, status: boolean): void;
+  clearWorkshop(workshopID: string): void;
+  progressListener: firebase.Unsubscribe | any; //find a way to get rid of any,
+  workshopListener: firebase.Unsubscribe | any; //find a way to get rid of any,
+  signOut(): void;
+  dataLoaded: boolean;
+  alert: boolean;
+  alertText: string;
+  resetAlertStatus(): void;
+}
+
+interface AdminDashboardState {
+  workshops: workshop[];
+  dataLoaded: boolean;
+  cards: any[];
```

Comment:
There's an interface for this defined in the `CardTile.tsx` components, we need to use the same one here